src/ripple/core/tests/Config.test.cpp

```diff
@@ -138,28 +128,26 @@ backend=sqlite
 class ConfigGuard
 {
 private:
+    bool rmSubDir_{false};
+
+protected:
     using path = boost::filesystem::path;
     path subDir_;
-    path configFile_;
-    path dataDir_;
 
-    bool rmSubDir_{false};
-    bool rmDataDir_{false};
-
-    Config config_;
+    auto rmDir (path const& toRm)
+    {
+        if (is_directory (toRm) && is_empty (toRm))
+            remove (toRm);
+        else
+            std::cerr << "Expected " << toRm.string ()
+                      << " to be an empty existing directory.\n";
```

Comment:
Even if I make `ConfigGuard` inherit from `beast::unit_test::suite`, using `log` doesn't seem to work here since the class's `run` function would not be called via `BEAST_DEFINE_TESTSUITE`.

I get a segfault with the `log` with `suite_` being null
https://github.com/ripple/rippled/blob/develop/src/beast/extras/beast/unit_test/suite.hpp#L275-L281